package/MDAnalysis/core/selection.py

```diff
@@ -217,6 +217,21 @@ def apply(self, group):
         return func(self, group)
     return apply
 
+
+def return_empty_on_empty_selection(func):
+    """
+    Decorator to return empty AtomGroups from the apply() function
+    if entire the selection is empty or a sub-expression is empty.
+    """
+    @functools.wraps(func)
+    def apply(self, group):
+        if len(group) == 0:
```

Comment:
If you stack decorators as I commented above, you could avoid replicating the functionality of `return_empty_on_apply`.